### ISSUE TITLE
Error with hpIncoming

### ISSUE THREAD

**Issue by rayfish** [kirk24788/KPS#16]
I am working on the restoration druid rotation. It already works quite well solohealing itself on the tanking dummy in the garrison.

When I enter a party or raid, the rotation doesn't target anything but spams this message.

`events.lua:163 - Error Interface\AddOns\KPS\modules\unit\unit.lua:37: Unknown Unit-Property 'hpInc'! on OnUpdate function function: 0x1becab9d0
`

Examples of possibly relevant conditions in the rotation are:

`kps.heal.defaultTarget.hpIncoming < 0.5`

It also makes use of `kps.heal.defaultTank`and `kps.heal.defaultTarget`.

I have tried to put my restoration rotation in my fork using the github webinterface. I'm not familiar with git/github/versioning in general and not sure if it worked. 
For some reason it only lets me edit one file per commit. If I'm doing anything wrong or annoying please tell me and I'll stop.

**Comment by kirk24788** [comment]
Do you still get those with your current rotation?
```events.lua:163 - Error Interface\AddOns\KPS\modules\unit\unit.lua:37: Unknown Unit-Property 'hpInc'! on OnUpdate function function: 0x1becab9d0 ```

It looks as if "xxx.hpInc" was used instead of "xxx.hpIncoming" - currently your rotation looks fine.


Regarding git - if you want to learn something about it, maybe the GitHub guide can help you: https://try.github.io/levels/1/challenges/1
Via the WebInterface you'll always be restricted to one file/commit, I would suggest you edit on your checked out copy and use a GUI tool, like [SourceTree](https://www.sourcetreeapp.com).

**Comment by kirk24788** [comment]
And no - it's not annoying, please keep on opening issues if you find anything.

**Comment by rayfish** [comment]
Yes I found the problem. The rotation kind of works now. 
It doesn't perform very well yet because the KPS healing target logic is very different from the old JPS one. 

In the BGs I've tested it in so far it barely (never) targets anybody else and only heals the player itself. I think this is in part due to `kps.RaidStatus.prototype.lowestInRaid` and `kps.RaidStatus.prototype.lowestTankInRaid` defaulting to the player if no other target is found.

I also think `kps.RaidStatus.prototype.defaultTarget` `kps.RaidStatus.prototype.defaultTank` need to be limited to players in range. 
I'll play around with my fork, compare the logic to the old JPS and see if I can get it to work like it did.

What's nice is that the JPS healing rotation used to cut my FPS in half and KPS doesn't seem to have any effect at all, I assume because it's just more efficient through caching. 👍
